src/test/integration/mocks/BeaconChainMock.t.sol

```diff
@@ -0,0 +1,689 @@
+// SPDX-License-Identifier: BUSL-1.1
```

Comment:
Nit: none of our mock files end in `.t.sol`

Reply:
This is true, but I think maybe they should... though I'm not entirely sure what `.t.sol` actually does after looking into it.

I asked in the foundry support channel and our slack: https://eigenlabs.slack.com/archives/C032198EJ23/p1701271995541119

Reply:
going to keep this as is for now, but happy to revisit later if we want to standardize